src/main/java/org/opensearch/jobscheduler/transport/request/ExtensionJobActionRequest.java

```diff
@@ -27,7 +34,63 @@
      * @throws IOException if serialization fails
      */
     public ExtensionJobActionRequest(String extensionActionName, T actionParams) throws IOException {
-        super(extensionActionName, JobDetailsService.convertParamsToBytes(actionParams));
+        super(extensionActionName, convertParamsToBytes(actionParams));
+    }
+
+    /**
+     * Finds the index of the specified byte value within the given byte array
+     *
+     * @param bytes the byte array to process
+     * @param value the byte to identify index of
+     * @return the index of the byte value
+     */
+    private static int indexOf(byte[] bytes, byte value) {
+        for (int offset = 0; offset < bytes.length; ++offset) {
+            if (bytes[offset] == value) {
+                return offset;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Trims off the fully qualified request class name bytes and null byte from the ExtensionActionRequest requestBytes
+     *
+     * @param requestBytes the request bytes of an ExtensionActionRequest
+     * @return the trimmed array of bytes
+     */
+    public static byte[] trimRequestBytes(byte[] requestBytes) {
+        int nullPos = indexOf(requestBytes, ExtensionJobActionRequest.UNIT_SEPARATOR);
```

Comment:
Note that `nullPos` was left over from when this was a null byte, and maybe should just be a `pos`.